astroid/node_classes.py

```diff
@@ -1025,70 +1530,235 @@ def _format_args(args, defaults=None, annotations=None):
+        """
         self.targets = targets
         self.value = value
 
 
 class AnnAssign(mixins.AssignTypeMixin, Statement):
-    """Class representing an AnnAssign node"""
+    """Class representing an :class:`ast.AnnAssign` node.
+
+    An :class:`AnnAssign` is an assignment with a type annotation.
+
+    >>> node = astroid.extract_node('variable: List[int] = range(10)')
+    >>> node
+    <AnnAssign l.1 at 0x7effe1d4c630>
+    """
 
     _astroid_fields = ('target', 'annotation', 'value',)
     _other_fields = ('simple',)
     target = None
+    """What is being assigned to.
+
+    :type: NodeNG or None
+    """
     annotation = None
+    """The type annotation of what is being assigned to.
+
+    :type: NodeNG
+    """
     value = None
+    """The value being assigned to the variables.
+
+    :type: NodeNG or None
+    """
     simple = None
+    """Whether :attr:`target` is a pure name or a complex statement.
+
+    :type: int
+    """
 
     def postinit(self, target, annotation, simple, value=None):
+        """Do some setup after initialisation.
+
+        :param target: What is being assigned to.
+        :type target: NodeNG
+
+        :param annotation: The type annotation of what is being assigned to.
+        :type: NodeNG
+
+        :param simple: Whether :attr:`target` is a pure name
+            or a complex statement.
+        :type simple: int
+
+        :param value: The value being assigned to the variables.
+        :type: NodeNG or None
+        """
         self.target = target
         self.annotation = annotation
         self.value = value
         self.simple = simple
 
 
 class AugAssign(mixins.AssignTypeMixin, Statement):
-    """class representing an AugAssign node"""
+    """Class representing an :class:`ast.AugAssign` node.
+
+    An :class:`AugAssign` is an assignment paired with an operator.
+
+    >>> node = astroid.extract_node('variable += 1')
+    >>> node
+    <AugAssign l.1 at 0x7effe1db4d68>
+    """
     _astroid_fields = ('target', 'value')
     _other_fields = ('op',)
     target = None
+    """What is being assigned to.
+
+    :type: NodeNG or None
+    """
     value = None
+    """The value being assigned to the variable.
+
+    :type: NodeNG or None
+    """
 
     def __init__(self, op=None, lineno=None, col_offset=None, parent=None):
-        self.op = op
-        super(AugAssign, self).__init__(lineno, col_offset, parent)
+        """
+        :param op: The operator that is being combined with the assignment.
+            This includes the equals sign.
+        :type op: str or None
+
+        :param lineno: The line that this node appears on in the source code.
+        :type lineno: int or None
+
+        :param col_offset: The column that this node appears on in the
+            source code.
+        :type col_offset: int or None
+
+        :param parent: The parent node in the syntax tree.
+        :type parent: NodeNG or None
+        """
+        self.op = op
```

Comment:
Initially I thought this was removed! But it was just lost through these docstrings. I think it would help if we could add an empty line between these blocks (attribute + docstring + newline). What do you think?

Reply:
I could put the docstring for the attribute instead. That's maybe a bit more conventional, and it won't spread out the code too much?